flink/src/test/java/org/apache/iceberg/flink/TestFlinkCatalogDatabase.java

```diff
@@ -0,0 +1,241 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink;
+
+import java.io.File;
+import java.util.Map;
+import org.apache.flink.table.catalog.exceptions.DatabaseNotEmptyException;
+import org.apache.iceberg.AssertHelpers;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableSet;
+import org.apache.iceberg.types.Types;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.Test;
+
+public class TestFlinkCatalogDatabase extends FlinkCatalogTestBase {
+
+  public TestFlinkCatalogDatabase(String catalogName, String[] baseNamepace) {
+    super(catalogName, baseNamepace);
+  }
+
+  @After
+  public void clean() {
+    sql("DROP TABLE IF EXISTS %s.tl", flinkIdentifier);
+    sql("DROP DATABASE IF EXISTS %s", flinkIdentifier);
+  }
+
+  @Test
+  public void testCreateNamespace() {
+    Assert.assertFalse(
+        "Database should not already exist",
+        validationNamespaceCatalog.namespaceExists(icebergNamespace));
+
+    sql("CREATE DATABASE %s", flinkIdentifier);
+
+    Assert.assertTrue("Database should exist", validationNamespaceCatalog.namespaceExists(icebergNamespace));
+  }
+
+  @Test
+  public void testDefaultDatabase() {
+    sql("USE CATALOG %s", catalogName);
+
+    Assert.assertEquals("Should use the current catalog", tEnv.getCurrentCatalog(), catalogName);
+    Assert.assertEquals("Should use the configured default namespace", tEnv.getCurrentDatabase(), "default");
+  }
+
+  @Test
+  public void testDropEmptyDatabase() {
+    Assert.assertFalse(
+        "Namespace should not already exist",
+        validationNamespaceCatalog.namespaceExists(icebergNamespace));
+
+    sql("CREATE DATABASE %s", flinkIdentifier);
+
+    Assert.assertTrue("Namespace should exist", validationNamespaceCatalog.namespaceExists(icebergNamespace));
+
+    sql("DROP DATABASE %s", flinkIdentifier);
+
+    Assert.assertFalse(
+        "Namespace should have been dropped",
+        validationNamespaceCatalog.namespaceExists(icebergNamespace));
+  }
+
+  @Test
+  public void testDropNonEmptyNamespace() {
+    Assume.assumeFalse("Hadoop catalog throws IOException: Directory is not empty.", isHadoopCatalog);
```

Comment:
This sounds like a bug in the Hadoop catalog. Can we fix it instead of ignoring this test case?

Reply:
I'll modify in this PR. Tell me if I need create a new PR for fixing hadoop catalog.

Reply:
I think it would be better to fix the Hadoop catalog in a separate PR and leave this one with the Assume until it is merged.

Reply:
I'll create it.